DataManagementSystem/scripts/dirac-dms-protocol-matrix.py

```diff
@@ -0,0 +1,173 @@
+"""
+  Generate a matrix of protocols used between SEs for FTS transfers.
+  The output is a CSV file containing a matrix source/destination.
+  The value of each cell is of the form "proto1/proto2 ([protoA,protoB,...])"
+  proto1/proto2 are the protocols that would really be given to FTS for source and dest urls
+  protoA,protoB,etc are the sorted list of protocols that would be attempted for thrid party copy by DIRAC
+
+  By default, all the SEs are taken into account, but the matrix is factorized by using baseSEs.
+
+  Suppose you have the following in your CS::
+
+    StorageElementBases{
+      IN2P3-Disk
+    }
+    StorageElements{
+      IN2P3-DST{
+        BaseSE = IN2P3-Disk
+      }
+      IN2P3-User{
+        BaseSE = IN2P3-Disk
+      }
+      AnotherDisk{
+      }
+    }
+
+  You can have the following combinations::
+
+    DIRAC-PROD>python ~/DIRAC/DataManagementSystem/scripts/dirac-dms-protocol-matrix
+    Using sources: IN2P3-Disk, AnotherDisk
+    Using target: IN2P3-Disk, AnotherDisk
+
+    DIRAC-PROD>python ~/DIRAC/DataManagementSystem/scripts/dirac-dms-protocol-matrix --FromSE=IN2P3-User
+    Using sources: IN2P3-User
+    Using target: IN2P3-Disk, AnotherDisk
+
+    DIRAC-PROD>python ~/DIRAC/DataManagementSystem/scripts/dirac-dms-protocol-matrix --FromSE=IN2P3-User --Bidirection
+    Using sources: IN2P3-User
+    Using target: IN2P3-User
+
+
+
+
+"""
+import csv
+from collections import defaultdict
+
+from DIRAC.Core.Base import Script
+Script.registerSwitch('', 'FromSE=', 'SE1[,SE2,...]')
+Script.registerSwitch('', 'TargetSE=', 'SE1[,SE2,...]')
+Script.registerSwitch('', 'OutputFile=', 'CSV output file (default /tmp/protocol-matrix.csv)')
+Script.registerSwitch('', 'Bidirection', 'If FromSE or TargetSE are specified, make a square matrix ')
+Script.setUsageMessage('\n'.join([__doc__,
+                                  'Usage:',
+                                  ' %s [option|cfgfile]  % Script.scriptName']))
+
+
+if __name__ == '__main__':
+  from DIRAC.Core.Base.Script import parseCommandLine
+  parseCommandLine()
+  from DIRAC import gConfig, gLogger
+  from DIRAC.DataManagementSystem.Utilities.DMSHelpers import DMSHelpers
+  from DIRAC.Resources.Storage.StorageElement import StorageElement
+
+  fromSE = []
+  targetSE = []
+  outputFile = '/tmp/protocol-matrix.csv'
+  bidirection = False
+  for switch in Script.getUnprocessedSwitches():
+    if switch[0] == 'FromSE':
+      fromSE = switch[1].split(',')
+    elif switch[0] == 'TargetSE':
+      targetSE = switch[1].split(',')
+    elif switch[0] == 'OutputFile':
+      outputFile = switch[1]
+    elif switch[0] == 'Bidirection':
+      bidirection = True
+
+  print fromSE
```

Comment:
Use the print function or logging (otherwise the py3k test in integration will start failing)

Reply:
Oups was not meant to be there anymore, I remove them